src/proto/connection.rs

```diff
@@ -51,6 +59,18 @@ where
     _phantom: PhantomData<P>,
 }
 
+struct DynConnection<'a, B: Buf = Bytes> {
```

Comment:
Dyn in Rust usually means dynamic dispatch. It looks like this is just to borrow multiple fields at once, right? Maybe instead something with the name `Ref`, like `ConnectionInnerRef`?

Reply:
It also replaces the `P: Peer` type parameter with an enum field. There was already `peer::Dyn` so I modeled the name on that (thereby `fn as_dyn`, instead of `fn as_ref` which is confusable with `AsRef::as_ref`).

Reply:
Oh right, I see now. Alright, seems fine.